tools/mo/openvino/tools/mo/main.py

```diff
@@ -310,24 +319,45 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    fallback_reasons['transformations_config'] = \
+        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
+
+    reasons = [reason for reason, is_applicable in fallback_reasons.items() if is_applicable(argv)]
+    return reasons
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
 
+def prepare_ir(argv : argparse.Namespace):
+    argv = arguments_post_parsing(argv)
     t = tm.Telemetry()
     graph = None
     ngraph_function = None
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
-
-    if moc_front_end:
+    fallback_reasons = check_fallback(argv)
+    if moc_front_end and len(fallback_reasons) == 0:
```

Comment:
in case `use_new_frontend` we should exclude fallback. The fallback is applicable only if new frontend is default and `use_new_frontend` and `use_legacy_frontend` are not specified in cmd line

Reply:
It seems to make sense. I've changed the logic